include/aie/Dialect/AIEVec/IR/AIEVecOps.td

```diff
@@ -667,4 +667,29 @@ def AIEVec_MaxOp:
   let assemblyFormat = "$lhs `,` $rhs attr-dict `:` type($result)";
   let hasVerifier = 0;
 }
+
+def AIEVec_CmpOp:
+  AIEVec_Op<"cmp", [
+    Pure,
+    AllTypesMatch<["lhs", "rhs"]>
+  ]>,
+  Arguments<(ins AnyVector:$lhs, AnyVector:$rhs, DefaultValuedStrAttr<StrAttr, "">:$pred)>,
+  Results<(outs AnyUnsignedInteger:$result)> {
+  let summary = "AIE vector comparison";
+  let description = [{
+    Xilinx-specific intrinsic that compares operations between two input vectors.
+    The attribute predicate defines which type of comparison is
+    performed. The following comparisons are supported:
+
+    -   equal (mnemonic: `"eq"`)
+    -   not equal (mnemonic: `"ne"`)
+    -   less than (mnemonic: `"lt"`)
```

Comment:
For arith.cmpi, the operands are always Signless and the predicate specifies whether the operands are interpreted to be either signed, e.g., slt, or unsigned, e.g., ult.
I think that we need to follow the same scheme, otherwise the transform will not preserve semantics.